From 0c40bdebc6caa444ab7c5591560c282f777001ed Mon Sep 17 00:00:00 2001
From: Robby Cai <r63905@freescale.com>
Date: Wed, 11 Jun 2008 17:41:15 +0800
Subject: [PATCH 023/505] ENGR00079187 BugFix green bar if video height is not
 integer multiples of 16

Fix Bug:
see green bar on the top of video when video height is not integer multiples of 16

Signed-off-by: Robby Cai <r63905@freescale.com>
---
 test/mxc_vpu_test/dec.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/mxc_vpu_test/dec.c b/test/mxc_vpu_test/dec.c
index 61fd7b1..ec81005 100644
--- a/test/mxc_vpu_test/dec.c
+++ b/test/mxc_vpu_test/dec.c
@@ -676,8 +676,8 @@ decoder_parse(struct decode *dec)
 			initinfo.minFrameBufferCount);
 
 	dec->fbcount = initinfo.minFrameBufferCount;
-	dec->picwidth = initinfo.picWidth;
-	dec->picheight = initinfo.picHeight;
+	dec->picwidth = ((initinfo.picWidth + 15) & ~15);
+	dec->picheight = ((initinfo.picHeight + 15) & ~15);
 	if ((dec->picwidth == 0) || (dec->picheight == 0))
 		return -1;
 
-- 
1.8.0

